extensions/script/ExecuteScript.h

```diff
@@ -185,23 +124,11 @@ class ExecuteScript : public core::Processor {
 
   ScriptEngineFactory engine_factory_;
 #ifdef LUA_SUPPORT
-  std::unique_ptr<ScriptEngineQueue<lua::LuaScriptEngine>> script_engine_q_;
+  std::shared_ptr<utils::ResourceQueue<lua::LuaScriptEngine>> lua_script_engine_queue_;
```

Comment:
The utils::ResourceQueue must be a shared_ptr because the ResourceWrappers have weak_ptrs in them so they can return the resource (if the queue is still available).

Reply:
Thanks, makes sense.